ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -457,14 +461,31 @@ void execute_program(std::string operation, std::string fname, std::string args,
 
 void execute_program(std::string fname, std::string args, bool wait) { execute_program("open", fname, args, wait); }
 
+// converts a relative path to absolute if the path exists
+std::string filename_absolute(std::string fname) {
+  if (file_exists(fname) || directory_exists(fname)) {
+    wchar_t rpath[MAX_PATH];
+    tstring tstr_fname = widen(fname);
+    GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL); 
+    return shorten(rpath);
+  } else { return ""; }
```

Comment:
Generally, statements in braces get separated onto their own lines. In this case, the `else` isn't actually required. Also, it's inconsistent with your other method.